daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetInputter.scala

```diff
@@ -413,14 +416,23 @@ abstract class InfosetInputter
         } catch {
           case ex: NonTextFoundInSimpleContentException =>
             UnparseError(One(elem.erd.schemaFileLocation), Nope, ex.getMessage())
+          case uri: URISyntaxException =>
+            UnparseError(One(elem.erd.schemaFileLocation), Nope, uri.getMessage())
         }
       if (optNilled.isDefined && optNilled.get) {
         if (txt != null && txt != "") {
           UnparseError(One(elem.erd.schemaFileLocation), Nope, "Nilled simple element %s has content", erd.namedQName.toExtendedSyntax)
         }
       } else if (erd.outputValueCalcExpr.isEmpty) {
         val primType = elem.erd.optPrimType.get
-        val obj = primType.fromXMLString(txt)
+        val obj = try {
+          primType.fromXMLString(txt)
+        } catch {
+          case ipd: InvalidPrimitiveDataException =>
+            UnparseError(One(elem.erd.schemaFileLocation), Nope, ipd.getMessage)
+          case oor: OutOfRangeException =>
```

Comment:
Wonder if instead of OutOfRange it would just be eaiser to alwys throw InvalidPrimitivateDataExceptino, and the message just includes range information? Then fromXMLString really only ever needs to catch a single exception?

Reply:
Agreed.